pkg/controller/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -884,35 +885,74 @@ func (r *ReconcileAWSPrivateLink) reconcileHostedZone(awsClient *awsClient,
 
 	hzLog := logger.WithField("hostedZoneID", hostedZoneID)
 
-	endpointDNSName := vpcEndpoint.DnsEntries[0].DnsName
-	endpointDNSHostedZone := vpcEndpoint.DnsEntries[0].HostedZoneId
+	rSet, err := r.recordSet(awsClient.hub, apiDomain, vpcEndpoint)
+	if err != nil {
+		hzLog.WithField("vpcEndpoint", aws.StringValue(vpcEndpoint.VpcEndpointId)).
+			WithError(err).Error("error generating DNS records")
+		return modified, "", err
+	}
 
 	_, err = awsClient.hub.ChangeResourceRecordSets(&route53.ChangeResourceRecordSetsInput{
 		HostedZoneId: aws.String(hostedZoneID),
 		ChangeBatch: &route53.ChangeBatch{
 			Changes: []*route53.Change{{
-				Action: aws.String(route53.ChangeActionUpsert),
-				ResourceRecordSet: &route53.ResourceRecordSet{
-					Type: aws.String("A"),
-					AliasTarget: &route53.AliasTarget{
-						DNSName:              endpointDNSName,
-						HostedZoneId:         endpointDNSHostedZone,
-						EvaluateTargetHealth: aws.Bool(false),
-					},
-					Name: aws.String(apiDomain),
-				},
+				Action:            aws.String(route53.ChangeActionUpsert),
+				ResourceRecordSet: rSet,
 			}},
 		},
 	})
 	if err != nil {
-		hzLog.WithField("aliasDNSName", endpointDNSName).
-			WithField("aliasHostedZone", endpointDNSHostedZone).
-			WithError(err).Error("error add record to Hosted Zone for VPC Endpoint")
+		hzLog.WithError(err).Error("error adding record to Hosted Zone for VPC Endpoint")
```

Comment:
Does the returned error in this case contain enough information that we don't need to include, say, the `rSet` as a `%+v`?

Reply:
the error from ChangeResourceRecordSets should give all the info needed to understand the cause of the problem.
the type of record type is also ready global, and therefore that info isn't required to be included here.